packages/redis_client/lib/src/redis_client.dart

```diff
@@ -596,6 +561,181 @@ class RedisTimeSeriesTimestamp {
   final String value;
 }
 
+/// {@template redis_time_series_clock}
+/// An abstract class which represents the clock
+/// in the context of a redis time series instance.
+/// {@endtemplate}
+// ignore: one_member_abstracts
+abstract class RedisTimeSeriesClock {
+  /// {@macro redis_time_series_clock}
+  const RedisTimeSeriesClock();
+
+  /// Returns a timestamp representing the current moment.
+  RedisTimeSeriesTimestamp now();
+}
+
+/// {@template redis_time_series_server_clock}
+/// A [RedisTimeSeriesClock] that represents time on the server.
+/// {@endtemplate}
+class RedisTimeSeriesServerClock extends RedisTimeSeriesClock {
+  /// {@macro redis_time_series_server_clock}
+  const RedisTimeSeriesServerClock();
+  @override
+  RedisTimeSeriesTimestamp now() => const RedisTimeSeriesTimestamp._('*');
+}
+
+/// {@template redis_time_series_client_clock}
+/// A [RedisTimeSeriesClock] that represents time on the client.
+/// {@endtemplate}
+class RedisTimeSeriesClientClock extends RedisTimeSeriesClock {
```

Comment:
I wonder if all these things need a "redis" prefix.  I suspect it could be removed everywhere and importers who care can just import it as 'redis'.